### PR TITLE
Subdirect

### PR DESCRIPTION
**Pull request by covracer**
### Background and Links
* Django imports at the top level, like `allowedflare.LoginView`, are problematic because they depend on Django setup

### Changes and Testing
* Rename `allowedflare.allowedflare` to `allowedflare.core` for brevity
* Drop Django-specific imports from the top level. This is a breaking change. All users will need to update their imports as demonstrated in `demodj/settings.py`
* Fix warning for backslash dot in plain string `'\.'`
* Improve the example JupyterHub configuration
* Upgrade dependencies and move from `requirements.in` to `pyproject.toml` because of issues with the `djangorestframework-stubs[compatible-mypy]` extra seemingly forcing the oldest version to be chosen; the UV docs say "Note extras are not supported with the requirements.in format."
* Use uv in the container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,11 +3,27 @@ FROM python:3.12
 # TODO assert C.UTF8 and PYTHONUNBUFFERED are set correctly
 
 WORKDIR /srv
-RUN --mount=type=cache,target=/root/.cache pip install --upgrade pip-tools wheel
+
+COPY includes.sh .
+# hadolint ignore=SC2086
+RUN --mount=type=cache,target=/var/cache/apt \
+    rm /etc/apt/apt.conf.d/docker-clean \
```
Comment (darpham) [inline Dockerfile:11]:
```suggestion
﻿﻿    rm /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache \
```
I believe you need to create apt conf to ensure they'll get cached

File: Dockerfile [hunk 2]
```diff
@@ -3,11 +3,27 @@ FROM python:3.12
...
+    && source includes.sh \
+    && apt-get update \
+    && apt-get install --no-install-recommends --quiet --quiet --yes \
+    && nodejs $PACKAGES \
```
Comment (darpham) [inline Dockerfile]:
```suggestion
    && apt-get install --no-install-recommends --quiet --yes \
       nodejs $PACKAGES \
```

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
             - run: npm install --frozen-lockfile
-
             - uses: actions/setup-python@v5
               with:
                   python-version-file: .python-version
 
-            - run: pip install --upgrade pip uv
-              env:
-                  PIP_PROGRESS_BAR: 'off'
+            - run: |
```
Comment (jamesbraza) [inline .github/workflows/check.yaml:23]: I moved to `setup-uv`, if you are curious

```yaml
      - uses: astral-sh/setup-uv@v3
        with:
          enable-cache: true
      - run: uv python pin ${{ matrix.python-version }}
      - run: uv sync --python-preference=only-managed
```

File: pyproject.toml [hunk 4]
```diff
@@ -15,8 +15,33 @@ dynamic = ['version']
 name = 'allowedflare'
 readme = 'README.md'
 
+[project.optional-dependencies]
+precommit = [
```
Comment (jamesbraza) [inline pyproject.toml:19]: Doesn't `pre-commit` handle its deps itself? Or is this to enable local usage of all `pre-commit` tools

**Review by jamesbraza — APPROVED** [review]

File: pyproject.toml (hunk 4 above)
Comment (covracer) [reply pyproject.toml:19]: This is for system/local installation of tools like codespell which pre-commit will use.

File: .github/workflows/check.yaml (hunk 3 above)
Comment (covracer) [reply .github/workflows/check.yaml:23]: I've been seeing `uv sync` and `uv.lock` in the docs. I'll have to try them out.

**Pull request merged**